example/js/js_image_layer.bzl

```diff
@@ -0,0 +1,260 @@
+# for why this is needed.
+_LAUNCHER_TMPL = """
+export BAZEL_BINDIR=.
+source {executable_path}
+"""
+
+def _write_laucher(ctx, executable_path):
+    "Creates a call-through shell entrypoint which sets BAZEL_BINDIR to '.' then immediately invokes the original entrypoint."
+    launcher = ctx.actions.declare_file("%s_launcher.sh" % ctx.label.package)
+
+    ctx.actions.write(
+        output = launcher,
+        content = _LAUNCHER_TMPL.format(executable_path = executable_path),
+        is_executable = True,
+    )
+
+    return launcher
+
+def _runfile_path(ctx, file, runfiles_dir):
+    return "/".join([runfiles_dir, to_manifest_path(ctx, file)])
+
+def _should_include(destination, include, exclude):
+    included = include in destination or include == ""
+    excluded = exclude in destination and exclude != ""
+    return included and not excluded
+
+def _runfiles_impl(ctx):
+    default = ctx.attr.binary[DefaultInfo]
+
+    executable = default.files_to_run.executable
+    executable_path = "/".join([ctx.attr.root, executable.short_path])
+    original_executable_path = executable_path.replace(".sh", "_.sh")
+    launcher = _write_laucher(ctx, original_executable_path)
+
+    file_map = {}
+
+    if _should_include(original_executable_path, ctx.attr.include, ctx.attr.exclude):
+        file_map[original_executable_path] = executable
+
+    if _should_include(executable_path, ctx.attr.include, ctx.attr.exclude):
+        file_map[executable_path] = launcher
+
+    manifest = default.files_to_run.runfiles_manifest
+    runfiles_dir = "/".join([ctx.attr.root, manifest.short_path.replace(manifest.basename, "")[:-1]])
+
+    files = depset(transitive = [default.files, default.default_runfiles.files])
+
+    for file in files.to_list():
+        destination = _runfile_path(ctx, file, runfiles_dir)
+        if _should_include(destination, ctx.attr.include, ctx.attr.exclude):
+            file_map[destination] = file
+
+    # executable and launcher should not go into runfiles directory so we add it to files here
+    files = depset([executable, launcher], transitive = [files])
+
+    symlinks = []
+
+    # NOTE: symlinks is different than root_symlinks. See: https://bazel.build/rules/rules#runfiles_symlinks for distinction between
+    # root_symlinks and symlinks and why they have to be handled differently.
+    for symlink in default.data_runfiles.symlinks.to_list():
+        destination = "/".join([runfiles_dir, ctx.workspace_name, symlink.path])
+        if not _should_include(destination, ctx.attr.include, ctx.attr.exclude):
+            continue
+        if hasattr(file_map, destination):
+            file_map.pop(destination)
+        info = PackageSymlinkInfo(
+            target = "/%s" % _runfile_path(ctx, symlink.target_file, runfiles_dir),
+            destination = destination,
+            attributes = {"mode": "0777"},
```

Comment:
Why not `0700`?

Reply:
I don't know. this was the default in rules_pkg.

Reply:
We need to understand the code that we copy-paste.